src/core/geometry/qgsgeometry.cpp

```diff
@@ -1000,7 +1000,7 @@ QgsGeometry QgsGeometry::orientedMinimumBoundingBox( double &area, double &angle
       height = bounds.height();
     }
 
-    pt2 = pt1;
+    pt1 = pt2;
```

Comment:
Ouch, nice fix!

(Funny story -- I ran into this bug a while ago, and for the last 2 years I've had a set of 6 tabs open on my phone researching alternative OMBB algorithms which I planned to get to "some time". Thanks for allowing me to finally close these tabs :wink:  )

Reply:
Thanks. :)

I spent more than 4 hours looking for an alternative; I thought there was a bug in the function when the tested points were orthogonal. When I came back with my coffee... I saw this... :laughing:

Reply:
Well, you beat me by 1.98904836 years :smile: